models/issue.go

```diff
@@ -549,6 +549,11 @@ func (issue *Issue) ReplaceLabels(labels []*Label, doer *User) (err error) {
 		}
 	}
 
+	issue.Labels = nil
```

Comment:
But query operations will not be rollback if `sess.Commit` failed. I think we could reload them after commit succeed.

Reply:
We cannot query them outside of the session. There is no guarantee that they will get the correct result outside of the session. (They may get another later label change result instead.) They have to be queried within the same transaction.

If the transaction fails then the issue is in an incorrect state anyway and the rest of the page errors out with no web hook called.